hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -1411,6 +1405,30 @@ AzureBlobFileSystemStore getAbfsStore() {
     return abfsStore;
   }
 
+  @VisibleForTesting
+  void setAbfsStore(AzureBlobFileSystemStore abfsStore) {
+    this.abfsStore = abfsStore;
+  }
+
+  @VisibleForTesting
+  void createFileSystemIfNotExist(TracingContext tracingContext) throws IOException {
+    if (this.tryGetFileStatus(new Path(AbfsHttpConstants.ROOT_PATH), tracingContext) == null) {
+      try {
+        this.createFileSystem(tracingContext);
+      } catch (IOException ex) {
+        if (ex instanceof AzureBlobFileSystemException) {
+          checkException(null, (AzureBlobFileSystemException) ex,
+              AzureServiceErrorCode.FILE_SYSTEM_ALREADY_EXISTS);
+        } else if (ex.getCause() != null && ex.getCause() instanceof AzureBlobFileSystemException) {
```

Comment:
instanceof always returns false if source is null, so the null check here can be removed